src/types/inc/viewport.hpp

```diff
@@ -13,15 +13,19 @@ Author(s):
 
 #pragma once
 #include "../../inc/operators.hpp"
+#include "../../inc/til.h"
 
 namespace Microsoft::Console::Types
 {
-    struct SomeViewports;
+    class Viewport;
+
+    typedef til::some<Viewport, 4> SomeViewports;
 
     class Viewport final
     {
     public:
         ~Viewport() {}
+        Viewport() noexcept;

```

Comment:
interesting! I'm surprised this became necessary; perhaps it's something we could improve in `some`

Reply:
I've tried bypass it by using `SomeViewports result{Viewport::Empty x 4}` but that would change the meaning of `some`, which I assume `_used` represents the number of element that's actually used. So I'm stucked here.

Reply:
Unless Dustin has an idea of how to avoid this, I'm OK with the default constructor making an empty one. We're going to hopefully improve `Viewport` into a `til::rect` or something in the future anyway and a default constructor will almost certainly be warranted at that point in time.

Reply:
Oh, I see. No, yeah, having a default constructor that does the same thing as `Viewport::Empty` is _100%_ the correct thing to do- to the point where I'm wondering why we ended up doing it not-that-way :P

We should do two or three things here.

1. make the constructor constexpr
2. make it initialize to the same values as `Empty()` (`0, 0, -1, -1`)
3. _make it live in the header_

Reply:
4? make `Empty` just do this?

Reply:
I will take a look at it. But I don’t really get it why a default is needed for the array. Perhaps I just don’t get C++...

Reply:
Ah! Okay, so, `std::array<T, 4>` requires that storage be allocated for four things. Since array doesn't have a use counter, there must be a valid value of type `T` in every addressable index. Without a default constructor, the compiler cannot emit anything that would truly and safely initialize the values.

This is largely the same as `Viewports v[4]` -- each four viewports will be created and default-constructed using the default constructor.

Reply:
Vector doesn't need to make this guarantee because accessing the objects beyond the end of the "used" region is undefined behavior.

Reply:
Thanks Dustin. My admiration for you remains the same. My love for Java & C# just got a *lot* deeper.